docker-compose.yml

```diff
@@ -1,25 +1,33 @@
 services:
   mongodb:
-    image: mongo:latest
+    image: rhaoio/rhaoio-mongo-merkle:6731bb3e3b41f7f507920699
     attach: false
     network_mode: "host"
+    environment:
+      # Set this port if 27017 is already used by another service/mongodb instance
+      # Mostly useful if using network_mode: "host", as the port will be shared.
+      - MONGO_INITDB_PORT=27017
     # ports:
-    #   - "27017:27017"
+    #  - "27017:27017"
     volumes:
-      - ./mongo/db:/data/db
-      - ./mongod.conf:/data/configdb/mongod.conf
+      - mongodb_data:/data/db
     logging:
       driver: "json-file"
       options:
         max-size: "10m"
         max-file: "5"
-    command: --config /data/configdb/mongod.conf
+    command: --port 27017
     healthcheck:
-      test: echo 'db.runCommand("ping").ok' | mongosh localhost:27017/test --quiet
-      start_period: 5s
+      test: |
+        mongosh --port 27017 --quiet --eval '
+          const ping = db.adminCommand({ ping: 1 }).ok;
+          const init = db.init_status.findOne({ "_id": "init" }) != null;
+          if (ping && init) { quit(0) } else { quit(1) }
+        '
       interval: 30s
+      start_period: 20s
       timeout: 10s
-      retries: 3
+      retries: 10
```

Comment:
A point need consider that if it is timeout because of machine slowness, we are hard to estimation which number of retries we need.

Also if it timeout but in fact it is because the hard disk is real slow and still need time to do restore, will the restart of docker compose up fix it?

A work round is to set the healthcheck retrie to very big which means we always "wait everything ready", so just rely on the hoster to manually stop it to check if it is stucking.